util/testutil/workers/util.go

```diff
@@ -49,12 +45,20 @@ func runBuildkitd(ctx context.Context, conf *integration.BackendConfig, args []s
 	if err := os.Chown(filepath.Join(tmpdir, "tmp"), uid, gid); err != nil {
 		return "", nil, err
 	}
-
 	deferF.Append(func() error { return os.RemoveAll(tmpdir) })
 
+	cfgfile, err := integration.WriteConfig(append(conf.DaemonConfig, integration.WithOTELSocketPath(getTraceSocketPath(tmpdir))))
```

Comment:
`WithOTELSocketPath` doesn't need to be defined as public func in the `integration` package. It can just be defined here with a private function.